### DISCUSSION TITLE
WOW! What a great find on the nuances of prompt design. Must Watch!

### DISCUSSION THREAD

**Discussion by VStudioAI** [vladmandic/sdnext#1361] [General]
Guys, 

I came across a real gem on YT yesterday. Tokenized AI. https://youtu.be/YPVxA7wi_mA

It is primarily about mid journey but there is a lot of cross over here. This guy is probably the best prompt master there is. Some of the techniques and observations that he presents on this channel are truly astonishing.

I'm not sure there is a 1 to 1 translation between mid journey and stable diffusion, but you'll see what I mean when you watch the video. He has several and they are all exceptional.

Prompting doesnt work the way you think it does.

**Comment by vladmandic** [comment]
set `SD_PROMPT_DEBUG=1` before you start the server and watch the new prompt parser in action in your logs :)

**Reply by Rogal80** [reply]
Where do i need to put this command ?

**Reply by brunogcar** [reply]
for windows, create a file called something  like webui-startup.bat

```
@ECHO off

set SD_PROMPT_DEBUG=1
set COMMANDLINE_ARGS= 

call webui.bat
```

and use that to  launch SD instead  of webui.bat

------------------------------------------------------

Aditional info: with that you can pass ARGS to initiate SD for example easily 

--upgrade 
--debug
--safe
etc  

set COMMANDLINE_ARGS= --upgrade
or 
call webui.bat --upgrade

**Comment by monoinyo** [comment]
you might like test my prompt- https://github.com/Extraltodeus/test_my_prompt

 it will separate generations into an image for each block of text between commas. For each image it shows you what the generation would look like without that block of text as part of the prompt.

**Comment by brunogcar** [comment]
a few extensions to help  build prompts

https://github.com/DominikDoom/a1111-sd-webui-tagcomplete
https://github.com/Physton/sd-webui-prompt-all-in-one
https://github.com/some9000/StylePile

and one to help analyze the effects of each part on image

https://github.com/toriato/stable-diffusion-webui-daam

**Reply by monoinyo** [reply]
DAAM looks way cool, gotta try it out
